### PR TITLE
Cheatsheets for CM1025 Fundamentals of Computer Science

### PR DESCRIPTION
**Pull request by lamafab**
Hello

I'm currently working on a collection of cheatsheets, meant to be useful for studying. Preview:

* [(First-order) Predicate & Propositional Logic](https://github.com/lamafab/notes/blob/fcs-notes/level-4/fundamentals-of-computer-science/student-notes/fabio-lama/cheatsheet_predicate_propositional_logic.pdf)
* [Proofs](https://github.com/lamafab/notes/blob/fcs-notes/level-4/fundamentals-of-computer-science/student-notes/fabio-lama/cheatsheet_proofs.pdf)
* [Permutations & Combinatorics ](https://github.com/world-class/notes/blob/master/level-4/computational-mathematics/student-notes/fabio-lama/cheatsheet_probability_combinatorics.pdf) (covered in my CM module notes)
* [Automata Theory](https://github.com/lamafab/notes/blob/fcs-notes/level-4/fundamentals-of-computer-science/student-notes/fabio-lama/cheatsheet_automata_theory.pdf)
* [Formal Languages & Regular Expressions](https://github.com/lamafab/notes/blob/fcs-notes/level-4/fundamentals-of-computer-science/student-notes/fabio-lama/cheatsheet_formal_languages_regular_expressions.pdf)
* [Context-Free Languages](https://github.com/lamafab/notes/blob/fcs-notes/level-4/fundamentals-of-computer-science/student-notes/fabio-lama/cheatsheet_context_free_languages.pdf)
* [Turing Machines](https://github.com/lamafab/notes/blob/fcs-notes/level-4/fundamentals-of-computer-science/student-notes/fabio-lama/cheatsheet_turing_machines.pdf)
* [Recursion](https://github.com/lamafab/notes/blob/fcs-notes/level-4/fundamentals-of-computer-science/student-notes/fabio-lama/cheatsheet_recursion.pdf)
  * _NOTE_ that some topics discussed in the Algorithms I/II weeks were skipped given that those were somewhat redundant with the Algorithms and Data Structures I module. Some of the algorithms are worth it to be studied independently, however, such as the Gale-Shapely algorithm for stable matching, etc.
* NOTE: Complexity theory is not covered by the cheatsheets.

Feedback and corrections are welcome.

### CODE REVIEW

**Comment by lamafab** [comment]
@sglavoie Ready to merge. Thanks a lot!

**Comment by sglavoie** [comment]
Fantastic! Thanks for sharing your knowledge with us in such a useful and digestible format! I ended up merging your commits with my comment embedded into it by accident. That will be an eternal message of gratitude that'll be part of the commit history. 😄

File: level-4/fundamentals-of-computer-science/student-notes/fabio-lama/src/propositional_logic.adoc [hunk 1]
```diff
@@ -0,0 +1,38 @@
+= Cheatsheet - Propositional Logic
+Fabio Lama <fabio.lama@pm.me>
+:description: Module: CM1025 Fundamentals to Computer Science, started 25. October 2022
+:doctype: article
+:sectnums: 4
+:stem:
+
```
Comment (Gabriel-Gotti) [inline level-4/fundamentals-of-computer-science/student-notes/fabio-lama/src/propositional_logic.adoc]: i had my royalty rights used by someone that has been hacking into my system. it has created more headaches than i wanted to notice for everyone that has been watching it go on. i feel as an O.S. system with licensing for communication and internet servicing and Networking. I have fifteen people plus some riding on my system that i cant get out. they have stole more money from me than what they are aware of me being able to watch my buffer times about. i do need a little security while i keep my investigation on going...

Comment (lamafab) [reply level-4/fundamentals-of-computer-science/student-notes/fabio-lama/src/propositional_logic.adoc]: Uhm... what?

File: level-4/fundamentals-of-computer-science/student-notes/fabio-lama/src/propositional_logic.adoc [hunk 2]
```diff
@@ -55,3 +55,82 @@ And the following connectives:
...
+++++
+true vv true = true\
+true vv false = true\
+false vv true = true\
+false vv false = false
+++++
+
+=== Implication: stem:[->]
+
+[stem]
```
[the remark this answers is not in the document]
Comment (lamafab) [reply level-4/fundamentals-of-computer-science/student-notes/fabio-lama/src/propositional_logic.adoc]: I don't understand what you mean. Can you clarify please?

**Pull request merged**